debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorTask.java

```diff
@@ -117,7 +111,12 @@ public void start(Map<String, String> props) {
             }
 
             // Set up the task record queue ...
-            this.queue = new TaskRecordQueue(config, replicaSets.replicaSetCount(), running::get, recordSummarizer);
+            this.queue = new ChangeEventQueue.Builder<SourceRecord>()
```

Comment:
Would it be possible to use a common parameters to configure the queue, not ones defined be each connector?
Could builder support self-configuration to by passing a `Configuartion` object to it?

Reply:
> Would it be possible to use a common parameters to configure the queue, not ones defined be each connector?

Yes, probably. I've filed https://issues.jboss.org/browse/DBZ-585 to explore config parameter consolidation.

> Could builder support self-configuration to by passing a Configuartion object to it?

While doable once ^^^ has been done, I prefer the explicit approach of making apparent what options are required by a given class. Also it's easier for testing.

Reply:
ok, just a matter of opinion :-). I mean to support both options - so testing can easily use builder methods but at the same time we would reduce code duplication among connectors. Or eat least it would be nice to have a helper class that would build the queue from Configuration obejct that would be used by all connectors.